skimage/io/_io.py

```diff
@@ -135,24 +183,52 @@ def imsave(fname, arr, plugin=None, check_contrast=True, **plugin_args):
     and largest file size (default 75). This is only available when using
     the PIL and imageio plugins.
     """
-    if isinstance(fname, pathlib.Path):
-        fname = str(fname.resolve())
-    if plugin is None and hasattr(fname, 'lower'):
-        if fname.lower().endswith(('.tiff', '.tif')):
-            plugin = 'tifffile'
+    if plugin_args:
+        warnings.warn(
+            "Additional keyword arguments are deprecated since version "
+            "0.23 and will be removed in 0.25 (or later). To avoid this "
+            "warning, please do not use additional keyword arguments. "
+            "Use imageio or a similar package instead.",
+            category=FutureWarning,
+            stacklevel=2,
+        )
+
     if arr.dtype == bool:
         warn(
             f'{fname} is a boolean image: setting True to 255 and False to 0. '
             'To silence this warning, please convert the image using '
             'img_as_ubyte.',
-            stacklevel=2,
+            stacklevel=3,
         )
         arr = arr.astype('uint8') * 255
     if check_contrast and is_low_contrast(arr):
-        warn(f'{fname} is a low contrast image')
-    return call_plugin('imsave', fname, arr, plugin=plugin, **plugin_args)
+        warn(f'{fname} is a low contrast image', stacklevel=3)
+
+    with warnings.catch_warnings():
+        warnings.filterwarnings(
+            action="ignore",
+            message=".*Use imageio or a similar package instead.*",
+            category=FutureWarning,
+            module="skimage.io",
+        )
+        use_plugin = (
+            plugin is not DEPRECATED
+            or plugin_args
+            or plugin_order()["imsave"][0] != "imageio"
+        )
+        if use_plugin:
+            plugin = None if plugin is DEPRECATED else plugin
+            if isinstance(fname, pathlib.Path):
+                fname = str(fname.resolve())
+            if plugin is None and hasattr(fname, 'lower'):
+                if fname.lower().endswith(('.tiff', '.tif')):
+                    plugin = 'tifffile'
+            return call_plugin('imsave', fname, arr, plugin=plugin, **plugin_args)
 
+    return iio.imwrite(fname, arr)
 
+
+@_deprecate_plugin_function
 def imshow(arr, plugin=None, **plugin_args):
```

Comment:
`imshow` and similar don't have to be deprecated. I could try to wrap matplotlib in a similar manner to wrapping imageio in `im{save,read}`. However, before investing the work, I'll first wait if we decide that it's worth keeping them around...